src/react-native/05-intro-to-testing/intro-to-testing.md

```diff
@@ -2,23 +2,84 @@
 @parent learn-react-native 5
 @outline 3
 
-@description TODO
+@description Learn about testing solutions in React Native.
 
 @body
 
 ## Overview
 
 In this section, you will:
 
-- TODO
+- Learn about React Native Testing Library
+- See the distinctions of testing for React Native components
 
 ## Objective: Testing in React Native
 
-TODO
+How do we know when our code is working correctly? How do we know it’s still working correctly in the future after we make changes to it?
 
-### Concept TODO
+Testing helps by verifying that given certain inputs our code generates expected outputs. So far we’ve copied existing tests to prove that we’ve completed the exercise correctly, now let’s dive in and learn about how React Native testing is done.
 
-TODO
+### Introducing React Native Testing Library
+
+[React Native Testing Library (RNTL)](https://callstack.github.io/react-native-testing-library/docs/getting-started) is a set of tools that helps us write robust tests for our React Native applications. Unlike some other testing libraries that focus on the internal state and implementation details of components, RNTL emphasizes testing the behavior of components as experienced by the end-users.
+In order to test components it provides light utility functions using React Test Renderer; most testing libraries use the DOM, but since React Native doesn't use the DOM, the React Test Renderer is necessary.
+
+Akin to the original React Testing Library, RNTL's north star is:
+
+> The more your tests resemble the way your software is used, the more confidence they can give you.
+
+RNTL encourages tests that focus on how the user interacts with the application, rather than the implementation details of the components.
+
+### Rendering and verifying a component in a test
+
+OK, you’re convinced that RNTL is a good idea, how do you use it to create tests for your React components?
+
+Let’s take a look at an example: say we want to test a component we created named `EmailInputField` that renders a form field. We expect the component will generate XML like this:
+
+```html
```

Comment:
RN doesn't generate anything like this. Just put the RN JSX and don't worry about comparing it to html/dom/etc. Same comment applies in a few places.